### PR TITLE
ENH: FormHandler support column aggregations. See #145

### PR DESCRIPTION
**Pull request by pratapvardhan**
See #145

### CODE REVIEW

File: gramex/data.py [hunk 1]
```diff
@@ -829,29 +831,28 @@ def _filter_db(engine, table, meta, controls, args, source='select', id=[]):
         # Apply controls
         if '_by' in controls:
             by = _filter_groupby_columns(controls['_by'], colslist, meta)
-            if len(by) > 0:
-                query = query.group_by(*by)
```
Comment (pratapvardhan) [inline gramex/data.py]: For SQL, just removed `len(by) > 0` check.

File: gramex/data.py [hunk 2]
```diff
@@ -706,38 +706,40 @@ def _filter_frame(data, meta, controls, args, source='select', id=[]):
...
-                    col, agg, val = _filter_col(key, data.columns)
-                    if agg is not None:
-                        # Convert aggregation into a Pandas GroupBy agg function
-                        agg = agg.lower()
-                        agg = _frame_functions.get(agg, agg)
-                        agg_cols.append(key)
-                        if col in agg_dict:
-                            agg_dict[col].append(agg)
-                        else:
-                            agg_dict[col] = [agg]
```
Comment (pratapvardhan) [inline gramex/data.py]: Moved this chuck before `len(by) > 0` and `data.groupby(by)` to later.

**Comment by pratapvardhan** [comment]
Looks to  be unrelated failing scheduler test case.

**Comment by sanand0** [comment]
- Add test case for `{'_by': []}` without aggregations
- Add test case for `{'_by': [], '_c': []}` and `{'_by': [], '_c': ['']}`. Check that it does not throw an exception. Do not test return value

**Comment by pratapvardhan** [comment]
@sanand0 -- tests are added.

**Pull request merged**